pkg/labels/fake_applicator.go

```diff
@@ -102,21 +105,73 @@ func (app *fakeApplicator) GetMatches(selector labels.Selector, labelType Type)
 	return results, nil
 }
 
-func (app *fakeApplicator) WatchMatches(selector labels.Selector, labelType Type, quitCh chan struct{}) chan []Labeled {
+func (app *fakeApplicator) WatchMatches(selector labels.Selector, labelType Type, quitCh <-chan struct{}) chan []Labeled {
 	ch := make(chan []Labeled)
+	timer := time.NewTimer(time.Duration(0))
 	go func() {
 		for {
+			select {
+			case <-quitCh:
+				return
+			case <-timer.C:
+			}
+			// upper bound on request rate
```

Comment:
Unit tests execute in a variety of environments and should not rely on timing for correctness. If you want to make sure that a group of changes are found by this method, you should add different synchronization here.

However, I suspect that what you really want is to test a consumer of the `chan []Labeled`. If so, your unit tests should construct the `[]Labeled` batches you want your consumer to see and send them over a channel you control.